engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/compile/nodecompilation/NodeCompiler.scala

```diff
@@ -125,6 +162,75 @@ class NodeCompiler(definitions: ProcessDefinition[ObjectWithMethodDef],
     NodeCompilationResult(expressionTypingInfo, None, newCtx, validParams)
   }
 
+  def compileFields(fields: List[graph.variable.Field], ctx: ValidationContext)
+                   (implicit nodeId: NodeId): NodeCompilationResult[List[compiledgraph.variable.Field]] = {
+    val compilations: ValidatedNel[ProcessCompilationError, List[(FieldExpressionTypingResult, TypedExpression)]] = fields.map { field =>
+      objectParametersExpressionCompiler
+        .compile(field.expression, Some(field.name), ctx, Unknown)
+        .map { typedExpression =>
+          val fieldExpressionTyping =
+            FieldExpressionTypingResult(field.name, ExpressionTypingResult(typedExpression.returnType, Some(typedExpression.typingInfo)))
+          (fieldExpressionTyping, typedExpression)
+      }
+    }.sequence
+
+    compilations match {
+      case Valid(compilation) =>
+        val expressionTypings = compilation.map(_._1)
+        val typedObj = TypedObjectTypingResult(
+          expressionTypings.map(et => (et.fieldName, et.typingResult)).toMap
+        )
+        val typingInfo = expressionTypings.flatMap(_.toExpressionTypingInfo).toMap
+        val compiledObject = compilation.map { case(fieldExprTyping, typedExpr) =>
+          compiledgraph.variable.Field(fieldExprTyping.fieldName, typedExpr.expression)
+        }
+        NodeCompilationResult(
```

Comment:
I have a feeling that we can introduce some helper methods for NodeCompilation result, but let's leave it for now ;)

Reply:
👍